src/io/flutter/run/bazel/BazelFields.java

```diff
@@ -88,60 +76,43 @@ public String getWorkingDirectory() {
     return null;
   }
 
-  /**
-   * Used only for deserializing old run configs.
-   */
-  @Deprecated
-  public void setWorkingDirectory(final @Nullable String workDir) {
-    if (entryFile == null && workDir != null) {
-      entryFile = workDir + "/lib/main.dart";
-    }
-  }
-
   @Nullable
-  public String getLaunchingScript() {
-    return launchScript;
-  }
-
-  public void setLaunchingScript(@Nullable String launchScript) {
-    this.launchScript = launchScript;
+  public String getBazelArgs() {
+    return bazelArgs;
   }
 
   @Nullable
   public String getAdditionalArgs() {
     return additionalArgs;
   }
 
-  public void setAdditionalArgs(@Nullable String additionalArgs) {
-    this.additionalArgs = additionalArgs;
-  }
 
   @Nullable
   public String getBazelTarget() {
     return bazelTarget;
   }
 
-  public void setBazelTarget(@Nullable String bazelTarget) {
-    this.bazelTarget = bazelTarget;
-  }
-
   public boolean getEnableReleaseMode() {
     return enableReleaseMode;
   }
 
-  public void setEnableReleaseMode(boolean enableReleaseMode) {
-    this.enableReleaseMode = enableReleaseMode;
-  }
-
   BazelFields copy() {
     return new BazelFields(this);
   }
 
-  @NotNull
-  BazelFields copyTemplateToNonTemplate(@NotNull final Project project) {
-    final Workspace workspace = WorkspaceCache.getInstance(project).getNow();
-    if (workspace == null) return new BazelFields(this);
-    return new BazelFields(this, workspace);
+  @Nullable
+  private String getLaunchScriptFromWorkspace(@NotNull final Project project) {
+    final Workspace workspace = getWorkspace(project);
+    String launchScript = workspace == null ? null : workspace.getLaunchScript();
+    if (launchScript != null) {
+      launchScript = workspace.getRoot().getPath() + "/" + launchScript;
+    }
+    return launchScript;
+  }
+
+  @Nullable
+  protected Workspace getWorkspace(@NotNull Project project) {
```

Comment:
This is dependency injection; the test version of the bazel fields is providing its own fake Workspace.

This is a code smell that suggests we need a standardized solution for DI in the Flutter plugin.

Reply:
Done, good idea.